### ISSUE TITLE
Another "Signin detected from a new device"issue

### ISSUE THREAD

**Issue by Flappes** [zammad/zammad#1389]
### Infos:

* Used Zammad version: 1.5.0-1504163322.7eedba7c.jessie
* Used Zammad installation source: debian package
* Operating system: debian 8.9
* Browser + version: IE 11 + FX 52.2.1 ESR

### Expected behavior:
Notification mails "Signin detected..." should able to be configured or at least switched off when logins are done via api-call
* 

### Actual behavior:
Anyone receives notification mail on every login via api-call. IP is shown as 127.0.0.1 (apachelogs show correct IP). When login to zammad directly, ip-addressinformation is correct in the notification mail.
There is no translation into german language, notification mail is always sent in english.
The link "manage your notification mail" leads into your profile which doesn't provide any setting about this signin detection. 

* 

### Steps to reproduce the behavior:
use zammad-api for login
*

**Comment by martini** [comment]
Hi @Flappes 

on question which is not clear to me. You get the notifications on every api call done by browser or done by using the api directly e. g. with php?

 -Martin

**Comment by Flappes** [comment]
Thanks a lot for the quick response.
I spoke to my colleague, who has developped our application (which uses api-calls to zammad for authentication, profile updates etc.) and tested for myself. 
Looks like that every api call done by browser generates a notification mail. i received three mails when doing a login, open my profile (which reads zammads user properties via api-call) and changed a value.

bye
Flappes

**Comment by martini** [comment]
@Flappes There are two issue which you can check if it's one of them:

1) Sending email on every API request e. g. via php if no user agent is given. This is fixed in upcoming next release. See https://github.com/zammad/zammad/issues/1208 - for workaround see https://github.com/zammad/zammad/issues/1313#issuecomment-319372133

2) Signin detected from a new device via same browser (looks like your description) https://github.com/zammad/zammad/issues/1337 -> Here it's important to know, if it's working correctly if you test with another browser? For more background details see https://github.com/zammad/zammad/issues/1337#issuecomment-321501065

**Comment by Flappes** [comment]
Case 1 fits here. 
Am i right that this will be fixed in version 2.0.0? What's the expected release date?
Isn't there any workaround possible, e.g. switching off the entire notification about logins?
And how do i change the text in the notification mail, e.g. to change the link for managing notification settings?

regards 
Flappes

**Comment by martini** [comment]
> Am i right that this will be fixed in version 2.0.0? What's the expected release date?

Soon. You can wait for it. 😻

> Isn't there any workaround possible, e.g. switching off the entire notification about logins?

Yes. See here https://github.com/zammad/zammad/issues/1313#issuecomment-319372133 (just remove not grand user_preferences.device to all roles will disable this feature at all).

> And how do i change the text in the notification mail, e.g. to change the link for managing notification settings?

You want do remove this "managing notification" from the email, right?

**Comment by Flappes** [comment]
No. I'm probably forced to change this link for customers, who will accessing Zammad notification through our own application, which has a different url. The value in zammad, which represents the url, needs to be left untouched, because it's required for internal access.

Currently zammad is held back in apt-get upgrade (2.0.0-1505116678.d6e5e1ea.jessie).
i assume from reading #1313 that signin-detection mails aren't send anymore when using Rest-api,
but still via https-logins. 
Can you state that your recommendation with revoking user preferences will be valid in the long run?
I assume that when using version 2.0.0 i can still disable Signing-Detection notification with your mentioned revokes?
I'm unsure if it comes with more trouble in future. 
Reading a statement to avoid direct database edits under every circumstances (read in the newest issue about version 2, which is unfortunately untrackable for me currently) gives me queasy feeling about that.

**Comment by hanneshal** [comment]
Hi @Flappes 

I'm sorry but @martini is a bit busy with the release. 
I read the whole thread and your last comment confused me ... :D

We fixed the login message bug #1208 for the API use when you don't pass an User-Agent and there is a way to disable the whole "new login" notification for existing users. This has to be done after the update to 2.0, one time for every user. #1313 Which is no direct database edit, it uses the same API as the UI does (RAILS), just on the console.  

```
rails c

admin = Role.find_by(name: 'Admin')
admin.permission_revoke('user_preferences')
admin.permission_grant('user_preferences.password')
admin.permission_grant('user_preferences.notifications')
admin.permission_grant('user_preferences.access_token')
admin.permission_grant('user_preferences.language')
admin.permission_grant('user_preferences.linked_accounts')
admin.permission_grant('user_preferences.avatar')
admin.permission_grant('user_preferences.calendar')

agent = Role.find_by(name: 'Agent')
agent.permission_revoke('user_preferences')
agent.permission_grant('user_preferences.password')
agent.permission_grant('user_preferences.notifications')
agent.permission_grant('user_preferences.access_token')
agent.permission_grant('user_preferences.language')
agent.permission_grant('user_preferences.linked_accounts')
agent.permission_grant('user_preferences.avatar')
agent.permission_grant('user_preferences.calendar')
```

What are you unsure about? I'm not really sure whats the issue right now.

regards
Johannes

**Comment by Flappes** [comment]
Sorry for the late reply.
As @martini wrote, that after upgrading to Zammad 2.0 one should restore the permission for the roles. I was interested in the reason, e.g. if there is a incompatibility existing after the upgrade or anything else
Second is, with zero knowledge in rails i don't get the syntax. 
But i trust you and MamoulianDelacroix that everything will be allright ;-)
